packages/core/src/utils/pipeHandler.ts

```diff
@@ -5,6 +5,34 @@ import { PlateEditor } from '../types/PlateEditor';
 import { DOMHandlers, HandlerReturnType } from '../types/plugin/DOMHandlers';
 import { TEditableProps } from '../types/slate-react/TEditableProps';
 
+export const convertDomEventToSyntheticEvent = (domEvent: Event) => {
```

Comment:
Did you try setting the return type here?

Reply:
Thanks for looking into this. I was able to find a solution that didn't involve any type assertions, which should ensure that future changes to React.SyntheticEvent will be flagged as needing to be changed here.